contracts/manager/DelegatedManager.sol

```diff
@@ -0,0 +1,473 @@
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+    SPDX-License-Identifier: Apache License, Version 2.0
+*/
+
+pragma solidity 0.6.10;
+
+import { Address } from "@openzeppelin/contracts/utils/Address.sol";
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { Ownable } from "@openzeppelin/contracts/access/Ownable.sol";
+import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+
+import { ISetToken } from "@setprotocol/set-protocol-v2/contracts/interfaces/ISetToken.sol";
+import { PreciseUnitMath } from "@setprotocol/set-protocol-v2/contracts/lib/PreciseUnitMath.sol";
+
+import { AddressArrayUtils } from "../lib/AddressArrayUtils.sol";
+import { IGlobalExtension } from "../interfaces/IGlobalExtension.sol";
+
+
+/**
+ * @title DelegatedManager
+ * @author Set Protocol
+ *
+ * Smart contract manager that maintains permissions and SetToken admin functionality via owner role. Owner
+ * works alongside methodologist to ensure business agreements are kept. Owner is able to delegate maintenance
+ * operations to operator(s). There can be more than one operator, however they have a global role so once
+ * delegated to they can perform any operator delegated roles. The owner is able to set restrictions on what
+ * operators can do in the form of asset whitelists. Operators cannot trade/wrap/claim/etc. an asset that is not
+ * a part of the asset whitelist, hence they are a semi-trusted party. It is recommended that the owner address
+ * be managed by a multi-sig or some form of permissioning system.
+ */
+contract DelegatedManager is Ownable {
+    using Address for address;
+    using AddressArrayUtils for address[];
+    using SafeERC20 for IERC20;
+
+    /* ============ Enums ============ */
+
+    enum ExtensionState {
+        NONE,
+        PENDING,
+        INITIALIZED
+    }
+
+    /* ============ Events ============ */
+
+    event MethodologistChanged(
+        address _newMethodologist
+    );
+
+    event ExtensionAdded(
+        address _extension
+    );
+
+    event ExtensionRemoved(
+        address _extension
+    );
+
+    event ExtensionInitialized(
+        address _extension
+    );
+
+    event OperatorAdded(
+        address _operator
+    );
+
+    event OperatorRemoved(
+        address _operator
+    );
+
+    event AllowedAssetAdded(
+        address _asset
+    );
+
+    event AllowedAssetRemoved(
+        address _asset
+    );
+
+    event UseAssetAllowlistUpdated(
+        bool indexed _status
```

Comment:
Should all these args actually be `indexed`? This looks like an odd one out.

Reply:
[833f4f6](https://github.com/SetProtocol/set-v2-strategies/pull/14/commits/833f4f609f10fa8ba7dde7959eaec98c13582524)